docs/tutorials/ai/durable-ai-agent/index.md

```diff
@@ -0,0 +1,6078 @@
+
+
+Next, the Worker object is created, passing in the `client`, the `task_queue`, the `activity_executor`, and then registering the individual Workflows and Activities the Worker can execute.
+The Worker is then started with `await worker.run()`, which creates a long-running process that will poll the Temporal service, executing Workflow and Activities when they are requested.
+
+Finally, the standard `if __name__ == "__main__"` calls the main function when you run `worker.py`, starting the Worker.
+
+Now that you have implemented your Worker, verify that it runs.
+
+### Testing the Worker
+
+Before starting the Worker, you need to start a Temporal service. 
+To start the local development server, open a terminal and run the following command:
+
+```bash
+temporal server start-dev
+```
+
+This starts a local Temporal service running on port 7233 with the web UI running on port 8233.
+The output of this command should resemble (The exact version numbers may not match):
+
+```output
+CLI 1.1.1 (Server 1.25.1, UI 2.31.2)
+
+Server:  localhost:7233
+UI:      http://localhost:8233
+Metrics: http://localhost:53697/metrics
+```
+
+Next, open another terminal and run your Worker:
+
+```bash
+uv run worker/worker.py
+```
+
+Your Worker should start, and the output should be:
+
+```output
+Worker will use LLM model: openai/gpt-4o
+Address: localhost:7233, Namespace default
+(If unset, then will try to connect to local server)
+AgentActivities initialized with LLM model: openai/gpt-4o
+Worker ready to process tasks!
+Starting worker, connecting to task queue: agent-task-queue
+Ready to begin processing...
+```
+
+The command will not exit, but will persist; this is expected.
+It is waiting for Workflows and Activity tasks to execute.
+If your Worker is running successfully, that's as much as you can test for the moment.
+Kill both the worker and Temporal service by pressing `CTRL-C` in each terminal.
+
+<details>
+
+<summary>
+Before moving on to the next section, verify that your files and directory structure are correct.
+</summary>
+
+```
+temporal-ai-agent/
+├── .env.example
+├── .gitignore
+├── .python-version
+├── README.md
+├── pyproject.toml
+├── uv.lock
+├── activities/
+|   ├── __init__.py
+|   └── activities.py
+├── models/
+│   ├── __init__.py
+│   ├── core.py
+│   └── requests.py
+├── prompts/
+│   ├── __init__.py
+│   ├── agent_prompt_generators.py
+│   └── prompts.py
+├── scripts/
+│   ├── create_invoice_test.py
+│   ├── find_events_test.py
+│   └── search_flights_test.py
+├── tools/
+│   ├── __init__.py
+│   ├── create_invoice.py
+│   ├── find_events.py
+│   ├── goal_registry.py
+│   ├── search_flights.py
+│   ├── tool_registry.py
+│   └── data/
+|       └── find_events_data.json
+├── worker/
+│   └── worker.py
+└── workflows/
+    ├── __init__.py
+    ├── agent_goal_workflow.py
+    └── workflow_helpers.py
+```
+
+</details>
+
+Next, you will implement a REST API that will serve as the backend service for invoking your agent.
```

Comment:
Even though this is like 75% of the way through I'd strongly consider stopping here with an example of how to interact with this thing purely using the command line and putting the UI in a followup tutorial. It's one of the only logical breakpoints we have.